src/metrics_server.rs

```diff
@@ -0,0 +1,170 @@
+use hyper::{
+    header::CONTENT_TYPE,
+    service::{make_service_fn, service_fn},
+    Body, Method, Request, Response, Server,
+};
+use prometheus::{Encoder, Gauge, IntCounter, IntGauge, Opts};
+use std::{convert::Infallible, env};
+
+pub type Result<T, E> = std::result::Result<T, E>;
+use chrono::{DateTime, NaiveDateTime, Utc};
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+
+pub fn try_create_int_counter(name: &str, help: &str) -> Result<IntCounter, prometheus::Error> {
+    let opts = Opts::new(name, help);
+    let counter = IntCounter::with_opts(opts)?;
+    prometheus::register(Box::new(counter.clone()))?;
+    Ok(counter)
+}
+
+pub fn try_create_int_gauge(name: &str, help: &str) -> Result<IntGauge, prometheus::Error> {
+    let opts = Opts::new(name, help);
+    let gauge = IntGauge::with_opts(opts)?;
+    prometheus::register(Box::new(gauge.clone()))?;
+    Ok(gauge)
+}
+
+pub fn try_create_gauge(name: &str, help: &str) -> Result<Gauge, prometheus::Error> {
+    let opts = Opts::new(name, help);
+    let gauge = Gauge::with_opts(opts)?;
+    prometheus::register(Box::new(gauge.clone()))?;
+    Ok(gauge)
+}
+
+lazy_static! {
+    pub static ref BLOCK_PROCESSED_TOTAL: IntCounter =
+        try_create_int_counter("total_blocks_processed", "Total number of blocks processed")
+            .unwrap();
+    pub static ref LATEST_BLOCK_HEIGHT: IntGauge =
+    try_create_int_gauge("latest_block_height", "Latest Block Height Of Indexer Thus Far")
+        .unwrap();
+    pub static ref LATEST_BLOCK_TIMESTAMP: Gauge =
+    try_create_gauge("latest_block_timestamp", "Timestamp of when last block was indexed.")
+        .unwrap();
+}
+
+enum ProbeResult {
+    IndexerStarted,
+    IndexerNotStarted,
+    SomethingIsWrong,
+}
+
+async fn serve_req(req: Request<Body>) -> Result<Response<Body>, hyper::Error> {
+    let response = match (req.method(), req.uri().path()) {
+        (&Method::GET, "/metrics") => {
+            let encoder = prometheus::TextEncoder::new();
+
+            let mut buffer = Vec::new();
+            if let Err(e) = encoder.encode(&prometheus::gather(), &mut buffer) {
+                eprintln!("could not encode custom metrics: {}", e);
+            };
+
+            let res = match String::from_utf8(buffer.clone()) {
+                Ok(v) => v,
+                Err(e) => {
+                    eprintln!("custom metrics could not be from_utf8'd: {}", e);
+                    String::default()
+                }
+            };
+            buffer.clear();
+
+            Response::builder()
+                .status(200)
+                .header(CONTENT_TYPE, encoder.format_type())
+                .body(Body::from(res))
+                .unwrap()
+        }
+        (&Method::GET, "/probe") => {
```

Comment:
health check for cloudrun. if a block was indexed within 2.5 mins of visiting `/probe`, this endpoint will return that the indexer is functioning normally or experiencing troubles.

Reply:
I've made it now so that it just shows the latest seen block time by the indexer as well as the timestamp difference between the last block and now. 

Instead of reporting here if the indexer is working fine, we can configure alerts from Grafana in case LATEST_BLOCK_TIMESTAMP_DIFF metric is > some amount of time.